Sources/Moho/Syntax.swift

```diff
@@ -32,18 +32,18 @@ public struct TypeSignature {
   let type: Expr
 }
 
-struct DeclaredClause {
-  let patterns: [DeclaredPattern]
-  let body: DeclaredClauseBody
+struct Clause {
+  let patterns: [Pattern]
+  let body: ClauseBody
 }
 
-enum DeclaredClauseBody {
-  case Empty
-  case Body(Expr, [Decl])
+enum ClauseBody {
+  case empty
+  case body(Expr, [Decl])
 }
 
 public indirect enum Expr: Equatable {
-  case lambda(Name, Expr)
+  case lambda(([Name], Expr), Expr)
```

Comment:
Would be very nice to comment these, and bonus: to add the type theory behind them in a quick comment for those of us unfamiliar with the dependently typed lambda calculus (read: me) 😛